File: ueueo-framework/ueueo-localization/src/main/java/Abp/Localization/LanguageInfo.cs
using System;
using JetBrains.Annotations;

namespace Volo.Abp.Localization;

[Serializable]
public class LanguageInfo : ILanguageInfo
{
    [NotNull]
    public   String CultureName { get; protected set; }

    [NotNull]
    public   String UiCultureName { get; protected set; }

    [NotNull]
    public   String DisplayName { get; protected set; }

    [CanBeNull]
    public   String FlagIcon;// { get; set; }

    protected LanguageInfo()
    {

    }

    public LanguageInfo(
        String cultureName,
        String uiCultureName = null,
        String displayName = null,
        String flagIcon = null)
    {
        ChangeCultureInternal(cultureName, uiCultureName, displayName);
        FlagIcon = flagIcon;
    }

    public   void ChangeCulture(String cultureName, String uiCultureName = null, String displayName = null)
    {
        ChangeCultureInternal(cultureName, uiCultureName, displayName);
    }

    private void ChangeCultureInternal(String cultureName, String uiCultureName, String displayName)
    {
        CultureName = Check.NotNullOrWhiteSpace(cultureName, nameof(cultureName));

        UiCultureName = !uiCultureName.IsNullOrWhiteSpace()
            ? uiCultureName
            : cultureName;

        DisplayName = !displayName.IsNullOrWhiteSpace()
            ? displayName
            : cultureName;
    }
}

File: ueueo-framework/ueueo-localization/src/main/java/com/ueueo/localization/LanguageInfo.java
package com.ueueo.localization;

import lombok.Data;
import org.apache.commons.lang3.StringUtils;

import java.util.Objects;

@Data
public class LanguageInfo implements ILanguageInfo {

    public String cultureName;

    public String uiCultureName;

    public String displayName;

    public String flagIcon;

    protected LanguageInfo() {

    }

    public LanguageInfo(String cultureName, String uiCultureName, String displayName, String flagIcon) {
        changeCultureInternal(cultureName, uiCultureName, displayName);
        this.flagIcon = flagIcon;
    }

    public void changeCulture(String cultureName, String uiCultureName, String displayName) {
        changeCultureInternal(cultureName, uiCultureName, displayName);
    }

    private void changeCultureInternal(String cultureName, String uiCultureName, String displayName) {
        Objects.requireNonNull(cultureName);
        this.cultureName = cultureName;
        this.uiCultureName = StringUtils.isNoneBlank(uiCultureName)
                ? uiCultureName
                : cultureName;
        this.displayName = StringUtils.isNoneBlank(displayName)
                ? displayName
                : cultureName;
    }
}
